src/DependencyInjection/M6WebGuzzleHttpExtension.php

```diff
@@ -72,7 +72,21 @@ protected function loadClient(ContainerBuilder $container, $clientId, array $con
         }
 
 
-
+        // process default headers if set
+        if (!empty($config['default_headers'])) {
+            $headers = [];
+            array_walk($config['default_headers'], function ($value, $key) use (&$headers) {
+                // replace underscore by hyphen in key
+                $key = preg_replace('`(?<!\\\)_`', '-', $key);
```

Comment:
with two backslashes you tell to regexp compilator to keep one underscore who escape the parenthesis  the third just say keep a backslash to escape next backslash. :)